src/core/instance/state.js

```diff
@@ -273,6 +273,11 @@ function initMethods (vm: Component, methods: Object) {
           `Avoid defining component methods that start with _ or $.`
         )
       }
+      if (typeof methods[key] !== 'function') {
```

Comment:
Could you make this test in development mode only and add a test (or/and adapt existing one), please?